app/assets/javascripts/Components/markus_file_manager.jsx

```diff
@@ -10,13 +10,21 @@ import { RawFileBrowser, Headers, FileRenderers, BaseFileConnectors } from 'reac
 
 
 class RawFileManager extends RawFileBrowser {
-  handleActionBarAddFileClick = (event) => {
+  handleActionBarAddFileClick = (event, selected) => {
     event.preventDefault();
-    window.modal_addnew.open();
+    let uploadTarget = '';
+    if (selected) {
+      if (selected.children) {
```

Comment:
Does this work if a folder is selected, but it has no children? (Is `selected.children` an empty list in this case?)

I believe the react-keyed-file-browser way of detecting whether something's a folder or not is whether its name ends with a `/`. Kind of weird, but I think maybe we should just be consistent with that.

Reply:
An empty directory still has a `children` attribute but I agree that keeping it consistent with the package behaviour makes sense.